src/compound_model/CompoundModelFactory.py

```diff
@@ -0,0 +1,403 @@
+            samples_to_gen, attr_sample_mode
+        )
+
+        # Parse input
+        parsed_input = self._controller.parse_native_input(random_input)
+
+        # Get latent codes
+        latent_codes = self._generator.random_latent_code(samples_to_gen)
+
+        # Generate images
+        uris, manipulated_latent_codes = self._controller.generate_native(
+            latent_codes, parsed_input
+        )
+
+        # Add images to population
+        self._population.add_all(
+            manipulated_latent_codes,
+            latent_codes,
+            uris,
+            [1] * len(uris),
+            append=(not replace),
+            **random_input,
+        )
+
+        # Clear compound metrics
+        self._clear_compound_metrics(self._population.get_name())
+
+    def _clear_compound_metrics(self, population_name: str) -> None:
+        # Clear compound metrics from manager
+        for filter in self._filters:
+            cmm = self._cmms[filter.get_name()]
+            if population_name == cmm.get_population().get_name():
+                cmm.clear_compound_metrics()
+
+        # Clear locally stored compound metrics from population
+        CompoundMetricManager.clear_local_compound_metrics(population_name)
+
+    def evaluate_compound_models(
+        self,
+        filters: Union[Type[Filter], list[Type[Filter]]] = None,
+        compound_metrics: Union[list[str], str] = None,
+        recalculate_metrics: bool = False,
+        **parameters: Any,
+    ) -> None:
+        """
+        Evaluates a selection or all of the compound models on a selection of compound metrics.
+        Result are saved to the managers storage.
+
+        Args:
+            filters (Union[Type[Filter], list[Type[Filter]]], optional): Selection of filters to use, must be subset
+                of the filters which was specified in the construction, see `get_filters()`.
+                `filters` can either be a filter, a list of filters or `None` which will result in the use of all available filters.
+                Defaults to None.
+            compound_metrics (Union[list[str], str], optional): List of compound metric names to calculate, must be subset
+                of the compound metrics which was specified in the construction, see `get_compound_metric_names()`.
+                `compound_metrics` can either be a compound metric name, a list of compound metric names or `None` which
+                will result in the use of all available compound metrics. Defaults to None.
+            recalculate_metrics (bool, optional): True if metrics should be recalculated even if they already exist.
+                False means that metrics already calculated will be skipped. Defaults to False.
+            **parameters(Any, optional): Parameters passed down to the calculation functions `calc()` and `get()`
+                of the compound metric manager. See their documentation for more details.
```

Comment:
do not put "optional" since this is implicit

maybe mention that the same parameters will be passed to all of the manager's underlying metrics

Reply:
Yee